be/src/agent/heartbeat_server.cpp

```diff
@@ -244,6 +245,18 @@ Status HeartbeatServer::_heartbeat(const TMasterInfo& master_info) {
         _engine.notify_listeners();
     }
 
+    if (master_info.__isset.meta_service_endpoint && config::meta_service_endpoint.empty()) {
```

Comment:
thrift server starts after `config::is_cloud_mode()` is called.
it is a bug that we leave `meta_servce_endpoint` and `cloud_unique_id` empty and change them after BE
 starts.
<img width="995" alt="image" src="https://github.com/user-attachments/assets/c00d3065-2db9-4c1b-8d4a-17d619500017">